### PR TITLE
Randomness Beacon for Immutable

### PR DESCRIPTION
**Pull request by end-r**

### CODE REVIEW

File: contracts/platform/__tests__/randomness.test.ts [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  describe('#commit', () => {
+
+    let beacon: Beacon;
+    let consumer: Consumer;
+
+    beforeEach(async() => {
+        beacon = await new BeaconFactory(provider.getSigner()).deploy();
+        consumer = await new ConsumerFactory(provider.getSigner()).deploy();
+    })
+
```
Comment (kermankohli) [inline contracts/platform/__tests__/randomness.test.ts]: Probably worth adding some basic checks for overflow. In general, the way to get the most bang for you buck when writing tests is to write negative test cases then test for positive.

File: contracts/platform/__tests__/randomness.test.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+  describe('#callback', () => {
+
+    let beacon: Beacon;
+    let consumer: Consumer;
+
+    beforeEach(async() => {
+        beacon = await new BeaconFactory(provider.getSigner()).deploy();
+        consumer = await new ConsumerFactory(provider.getSigner()).deploy();
+    });
+
```
Comment (kermankohli) [inline contracts/platform/__tests__/randomness.test.ts]: Same as above. It's good that we have tests that check if something works but not if the wrong inputs cause undesired state changes

File: contracts/platform/__tests__/randomness.test.ts [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        await beacon.recommit(receipt.blockNumber, 0);
+        await blockchain.waitBlocksAsync(256);
+        await beacon.recommit(receipt.blockNumber, 0);
+    });
+
+    it('should not be able to recommit twice too quickly', async () => {
+        let receipt = await beacon.commit(0);
+        await blockchain.waitBlocksAsync(256);
+        await beacon.recommit(receipt.blockNumber, 0);
+        await expectRevert(beacon.recommit(receipt.blockNumber, 0));
```
Comment (kermankohli) [inline contracts/platform/__tests__/randomness.test.ts]: Might be worth adding some expect statements around the events themselves being emitted

File: contracts/platform/contracts/randomness/Beacon.sol [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    constructor() public {
+        commitRequested.length = 2**256-1;
+    }
+
+    /**
+     * @dev Request randomness derived from a particular block
+     *
+     * @param offset the offset from the current block of the block which will be used in our random seed
+     */
+    function commit(uint256 offset) public returns (uint256) {
```
Comment (kermankohli) [inline contracts/platform/contracts/randomness/Beacon.sol]: Sorry for being super picky here but could we change the input names to begin with underscores? Happy to discuss this offline

File: contracts/platform/contracts/randomness/test/Consumer.sol [hunk 5]
```diff
@@ -0,0 +1,12 @@
+pragma solidity 0.5.11;
+
+import "../IBeacon.sol";
+
+contract Consumer {
+
+    function multiCommit(IBeacon beacon, uint count) public {
```
Comment (kermankohli) [inline contracts/platform/contracts/randomness/test/Consumer.sol]: What's the point of this contract exactly? Name doesn't make sense to me/I'm missing something here

File: contracts/platform/__tests__/randomness.test.ts [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+  describe('#recommit', () => {
+
+    let beacon: Beacon;
+    let consumer: Consumer;
+
+    beforeEach(async() => {
+        beacon = await new BeaconFactory(provider.getSigner()).deploy();
+        consumer = await new ConsumerFactory(provider.getSigner()).deploy();
+    });
+
```
Comment (kermankohli) [inline contracts/platform/__tests__/randomness.test.ts]: Would be good to stick with the subject pattern here so we can extend our input test cases if needed

File: contracts/platform/contracts/randomness/test/Consumer.sol (hunk 5 above)
Comment (end-r) [reply contracts/platform/contracts/randomness/test/Consumer.sol]: It's a consumer of the randomness beacon for testing purposes (it's in a test subdirectory to indicate that shouldn't be used externally). Thoughts on this pattern here?

**Pull request merged**